microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,562 @@
+            Http2Connection connection = new DefaultHttp2Connection(false);
+            Http2ConnectionHandler connectionHandler = new Http2ConnectionHandler(
+                    new DefaultHttp2ConnectionDecoder.Builder().connection(connection)
+                            .frameReader(new DefaultHttp2FrameReader()).listener(new Http2FrameAdapter()),
+                    new DefaultHttp2ConnectionEncoder.Builder().connection(connection).frameWriter(env.clientWriter));
+
+            @Override
+            public <T> Attribute<T> attr(AttributeKey<T> key) {
+                return null;
+            }
+
+            @Override
+            public <T> boolean hasAttr(AttributeKey<T> key) {
+                return false;
+            }
+
+            @Override
+            public Channel channel() {
+                return channel;
+            }
+
+            @Override
+            public EventExecutor executor() {
+                return executor;
+            }
+
+            @Override
+            public ChannelHandlerInvoker invoker() {
+                return null;
+            }
+
+            @Override
+            public String name() {
+                return "";
+            }
+
+            @Override
+            public ChannelHandler handler() {
+                return connectionHandler;
+            }
+
+            @Override
+            public boolean isRemoved() {
+                return false;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelRegistered() {
+                try {
+                    handler().channelRegistered(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelUnregistered() {
+                try {
+                    handler().channelUnregistered(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelActive() {
+                try {
+                    handler().channelActive(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireChannelInactive() {
+                try {
+                    handler().channelInactive(this);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return this;
+            }
+
+            @Override
+            public ChannelHandlerContext fireExceptionCaught(Throwable cause) {
+                try {
+                    handler().exceptionCaught(this, cause);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
I'm not too familiar with JMH but in Caliper System.err & System.out get redirected to /dev/null and none of these would ever show up. If an exception is being thrown in the pipeline and not caught the benchmark performance will degrade massively and give misleading results. Is there are an API to trigger benchmark failure for scenarios like this?